src/sync.js

```diff
@@ -621,21 +622,30 @@ SyncApi.prototype._sync = async function(syncOptions) {
         await client.store.setSyncData(data);
     }
 
+    const syncEventData = {
+        oldSyncToken: syncToken,
+        nextSyncToken: data.next_batch,
+        catchingUp: this._catchingUp,
+    };
+
+    if (this.opts.crypto && !isCachedResponse) {
+        // tell the crypto module we're about to process a sync
+        // response
+        await this.opts.crypto.onSyncWillProcess(syncEventData);
```

Comment:
Adds the callback for before the sync is processed - this is where we stop tracking all the devices on an initial sync, so we mark the appropriate ones as tracked again during the processing of the sync.